src/commands/aem/rde/install.js

```diff
@@ -192,13 +195,33 @@ class DeployCommand extends BaseCommand {
           });
         } else {
           type = guessedTypes[0];
+          this.doLog(chalk.yellow(`No --type provided, using ${type}`));
+        }
+      }
+
+      // when no types was defined explicitly, we try to guess the path as well
+      if ((type === 'content-file' || type === 'content-xml') && !flags.path) {
+        if (isLocalFile && inputPath.includes(JCR_ROOT)) {
+          const guessedPath = guessPath(inputPath);
+          if (guessedPath) {
+            flags.path = guessedPath;
+            this.doLog(
+              chalk.yellow(
+                `No --path provided, repository path was set to ${flags.path}`
+              )
+            );
+          }
```

Comment:
since path is used for the two types content-file and content-xml only, there's no difference in quality of guesses I would say? Can you please elaborate your usecase that you have in mind?

Reply:
If neither `--path` nor `--type` is set, but the `inputPath` contains the folder `jcr_root`, then we can infer that `--type` is one of `content-xml` or `content-file`, and we can also infer the `--path`.

So we can narrow down the candidate types based on the presence of `jcr_root` in the input file's local path, allowing for better guesses. Maybe I am missing something though.

Reply:
I read the code again, and I believe that my suggestion offers no improvement. Please forget about it 🙂

Reply:
:-)